### ISSUE TITLE
The_SSHLogger class in asyncssh logging module uses a mutable default argument 

### ISSUE THREAD

**Issue by t3lurid3** [ronf/asyncssh#275]
Relevant LOC https://github.com/ronf/asyncssh/blob/master/asyncssh/logging.py#L33

This creates a potential issue as `__package__` can be None. This means the logging module will uses the root logger instance which is generally not desired. It also means that changing the log level for asyncssh will also affect the log level for the root logger and any named loggers.

Similar issue reported here https://github.com/ronf/asyncssh/issues/156

I propose adding a small change to `_SSHLogger` class to prevent asyncssh using root logger instance inadvertently. 

This will save users having to decorate the asyncssh logging constructor to overwrite the
default argument value of the root logger.

As per your recommendation creating an issue before submitting a pull request.

**Comment by ronf** [comment]
I appreciate the report, but I'm not sure I see the problem you mention here. The _SSHLogger() class that you're referencing here is private to the asyncssh.logging module, and the specific assignment of the class variable _pkg_logger would always be evaluate in the context of that module, where __package__ should always be 'asyncssh', so _pkg_logger should always end up being a Logger object associated with the name 'asyncssh', and never the root logger.

Also, since _SSHLogger() is private to that module, the only instance of that class should actually be the "logger" object created within that module, and exported under the name asyncssh.logger.

I know there was a report in #156 of __package__ not working right with asyncssh distributed as an egg, but at the time I switched to distributing it as a wheel instead and that issue didn't seem to occur. Are you running into some other condition under which __package__ isn't working?

What sort of change did you have in mind?

**Comment by t3lurid3** [comment]
The issue is when `__package__` is None as it is when I use a wheel of asyncssh. As a simple safety measure we could for example add an additional check to specify that _pkg_logger use get a named logger either say:

```
class _SSHLogger(logging.LoggerAdapter):
    """Adapter to add context to AsyncSSH log messages"""

    _debug_level = 1
-    _pkg_logger = logging.getLogger(__package__)
+    _logger_name = __package__ or "asyncssh" 
+    _pkg_logger = logging.getLogger( _logger_name)
```

or

```
class _SSHLogger(logging.LoggerAdapter):
    """Adapter to add context to AsyncSSH log messages"""

    _debug_level = 1
    _pkg_logger = logging.getLogger(__package__)
+    if isinstance(_pkg_logger, logging.RootLogger):
+        _pkg_logger = logging.getLogger("asyncssh")
```

**Comment by ronf** [comment]
I can certainly add this guard, but I'm curious why you're seeing this, as I haven't run into this problem when installing the wheel that I publish on PyPI. Are you using the PyPI wheel, or getting it from somewhere else? What platform are you trying to install on? What version of wheel support were you using?

**Comment by pyhedgehog** [comment]
I bet it's from output of some code-analyzer. :-)

**Comment by ronf** [comment]
To be on the safe side, I've gone ahead and pushed commit 7af97f2 in the "develop" branch:

```patch
diff --git a/asyncssh/logging.py b/asyncssh/logging.py
index 0664a84..f26c82f 100644
--- a/asyncssh/logging.py
+++ b/asyncssh/logging.py
@@ -28,7 +28,7 @@ class _SSHLogger(logging.LoggerAdapter):
     """Adapter to add context to AsyncSSH log messages"""
 
     _debug_level = 1
-    _pkg_logger = logging.getLogger(__package__)
+    _pkg_logger = logging.getLogger(__package__ or 'asyncssh')
 
     def __init__(self, parent=_pkg_logger, child=None, context=''):
         self._context = context
```

I'll include this change in the next release.

I'd still be interested in knowing more about what would cause `__package__` to not be set correctly, though.

**Comment by t3lurid3** [comment]
I suspect a contributor to the issue may be a custom loader we use but will confirm when I have a chance to test.  I believe the change you have committed makes sense in any event as it ensures intended behaviour regardless of how downstream users build. Thanks.

**Comment by ronf** [comment]
Ah, that could be. Thanks for the report!

**Comment by ronf** [comment]
This change is now included in AsyncSSH 2.4.0.
